src/client.js

```diff
@@ -0,0 +1,442 @@
+          );
+
+          await sleep(delay);
+          return fetch(attempt);
+        } else {
+          if (err.response && err.response.body) {
+            debug('Error calling: %s NOT retrying! info: %j', entry.name, err.response.body);
+          } else {
+            debug('Error calling: %s NOT retrying! info: %j', entry.name, err);
+          }
+        }
+
+        throw err;
+      }
+    };
+
+    // Initialize the first request starting at 0 attempts
+    return fetch(0);
+  }
+
+  willRetry(err, attempts) {
+    const { retries } = this._options;
+    const { response } = err;
+
+    if (attempts > retries) {
+      return false;
+    }
+
+    if (!response) {
+      return true;
+    }
+
+    // If we got a response we read the error code from the response
+    return response.statusCode >= 500 && response.statusCode < 600;
+  }
+
+  async _request(entry, ...args) {
+    // Get number of arguments
+    const entryArity = entry.args.length + (entry.input ? 1 : 0);
+    // Get the query-string options taken
+    const optionKeys = entry.query || [];
+    const arity = args.length;
+
+    // Validate number of arguments
+    assert(arity === entryArity || (optionKeys.length && arity === entryArity + 1),
+      `Function \`${entry.name}\` expected ${entryArity} arguments but only received ${arity}`);
+
+    const endpoint = this.buildEndpoint(entry, args);
+    // Create url for the request
+    const url = this._options.baseUrl + endpoint;
+    // Add payload if one is given
+    const payload = entry.input ? args[entryArity - 1] : null;
+    // Find query string options if present
+    const query = args[entryArity] || {};
+
+    Object
+      .keys(query)
+      .forEach(key => assert(optionKeys.includes(key),
+        `Function \`${entry.name}\` expected options ${optionKeys.join(', ')} but received ${key}`));
+
+    const { monitor } = this._options;
+    const start = monitor ? process.hrtime() : null;
+
+    try {
+      const response = await this.makeRequest(entry, url, payload, query);
+
+      if (monitor) {
+        const end = process.hrtime(start);
+
+        monitor.measure([entry.name, 'success'], end[0] * 1000 + end[1] / 1000000);
+        monitor.count([entry.name, 'success']);
+      }
+
+      if (!response.headers['content-type'].includes('application/json') || !response.body) {
+        debug(`Empty response from server: call: ${entry.name}, method: ${entry.method}`);
+      } else {
+        return response.body;
+      }
+    } catch (err) {
+      if (monitor) {
+        const end = process.hrtime(start);
+        const code = err.response ? err.response.statusCode : err.code;
+        const state = code >= 500 ? 'server-error' : 'client-error';
+
+        monitor.measure([entry.name, state], end[0] * 1000 + end[1] / 1000000);
+        monitor.count([entry.name, state]);
+      }
+
+      if (!err.response) {
+        throw err;
+      }
+
+      const { response } = err;
+      const { body } = response;
```

Comment:
So because you set `options.json = true` for the request it parses the body... But if the body is invalid I don't want a JSON error.. I want an error with statusCode property on it... because most likely it's a 500 error from heroku because our app is down.